src/lib.rs

```diff
@@ -186,50 +193,170 @@ decl_module! {
 
 impl<T: Trait> Module<T> {
 
-    pub fn create_class(name: Vec<u8>, description: Vec<u8>, properties: Vec<Property>, schemas: Vec<ClassSchema>) -> Result<ClassId, &'static str> {
-        // TODO impl
+    pub fn create_class(
+        properties: Vec<Property>,
+        schemas: Vec<ClassSchema>,
+        name: Vec<u8>,
+        description: Vec<u8>
+    ) -> Result<ClassId, &'static str> {
+        
+        ensure!(schemas.len() > 0, ERROR_CLASS_EMPTY_SCHEMAS);
+        ensure!(properties.len() > 0, ERROR_CLASS_EMPTY_PROPS);
+        ensure!(name.len() > 0, ERROR_CLASS_EMPTY_NAME);
+        ensure!(description.len() > 0, ERROR_CLASS_EMPTY_DESCRIPTION);
+
+        let class_id = <NextClassId<T>>::get();
+
+        let new_class = Class {
+            id: class_id,
+            properties,
+            schemas,
+            name,
+            description,
+        };
+
+        // Save newly created class:
+        <ClassById<T>>::insert(class_id, new_class);
+
+        // Increment the next class id:
+        <NextClassId<T>>::put(class_id + 1);
```

Comment:
Use `mutate` instead, its semantically cleaner.

Reply:
Ok, changed to this:
```suggestion
        <NextClassId<T>>::mutate(|n| *n += 1);
```